pkg/symbol/addr2line/symtab.go

```diff
@@ -57,25 +57,29 @@ func Symbols(logger log.Logger, filename string, f *elf.File, demangler *demangl
 		logger:    log.With(logger, "liner", "symtab"),
 		searcher:  searcher,
 		demangler: demangler,
-		filename:  filename,
-		f:         f,
+		objFile:   objFile,
 	}, nil
 }
 
 func (lnr *SymtabLiner) Close() error {
-	return lnr.f.Close()
+	return lnr.objFile.ElfFile.Close()
 }
 
 func (lnr *SymtabLiner) File() string {
-	return lnr.filename
+	return lnr.objFile.Path
 }
 
 func (lnr *SymtabLiner) PCRange() ([2]uint64, error) {
 	return lnr.searcher.PCRange()
 }
 
 // PCToLines looks up the line number information for a program counter (memory address).
-func (lnr *SymtabLiner) PCToLines(addr uint64) (lines []profile.LocationLine, err error) {
+func (lnr *SymtabLiner) PCToLines(addr uint64, isRawAddr bool) (lines []profile.LocationLine, err error) {
+	if isRawAddr {
+		if addr, err = lnr.objFile.ObjAddr(addr); err != nil {
```

Comment:
We need to discuss whether we want to pull this logic from the agent to here. 

The object file we have here could be just a debuginfo file, so it's highly likely that it does not have the required `ProgramHeader` and `Note` sections. This is a design choice we made. We only want to send what is strictly necessary over the wire from the hosts.

We should send the necessary information (base address) as part of the API (probably in the mapping),

I'm happy to base address calculations as part of the tests. However, we can vastly simplify them. It's a shame that we cannot directly use these facilities from the agent (cyclic dependencies are hard).

Reply:
Also, with the recent changes, we moved most of the logic to the `Mapping` layer, which is semantically more correct. This information that is needed here belongs to the mapping not to the object file.

Reply:
Thank you! I will have a look at the recent changes in the agent.

Reply:
I've added base address fields to protobufs so they could be passed from the agent and stored in Badger, and later used during symbolization.